modules/contact/src/actions/ContactAction.C

```diff
@@ -368,72 +395,161 @@ ContactAction::addMortarContact()
+      }
     }
-
-    // Add the tangential Lagrange multiplier constraint on the secondary boundary.
-    if (_model == "coulomb")
+    // Add the tangential and normal Lagrange's multiplier constraints on the secondary boundary.
+    else if (_model == "coulomb")
     {
-      InputParameters params = _factory.getValidParams("TangentialMortarLMMechanicalContact");
-
-      params.set<BoundaryName>("primary_boundary") = _primary;
-      params.set<BoundaryName>("secondary_boundary") = _secondary;
-      params.set<SubdomainName>("primary_subdomain") = primary_subdomain_name;
-      params.set<SubdomainName>("secondary_subdomain") = secondary_subdomain_name;
-      params.set<NonlinearVariableName>("variable") = tangential_lagrange_multiplier_name;
-      params.set<bool>("use_displaced_mesh") = true;
-      params.set<MooseEnum>("ncp_function_type") = "fb";
-      params.set<Real>("c") = getParam<Real>("c_tangential");
-      params.set<bool>("compute_primal_residuals") = false;
-      params.set<NonlinearVariableName>("contact_pressure") = normal_lagrange_multiplier_name;
-      params.set<Real>("friction_coefficient") = getParam<Real>("friction_coefficient");
-
-      params.set<VariableName>("secondary_variable") = displacements[0];
-      if (ndisp > 1)
-        params.set<NonlinearVariableName>("secondary_disp_y") = displacements[1];
-      // secondary_disp_z is not implemented for tangential (yet).
-
-      _problem->addConstraint(
-          "TangentialMortarLMMechanicalContact", action_name + "_tangential_lm", params);
+      if (_mortar_approach == MortarApproach::Weighted)
+      {
+        InputParameters params =
+            _factory.getValidParams("ComputeFrictionalForceLMMechanicalContact");
+
+        params.set<BoundaryName>("primary_boundary") = _primary;
+        params.set<BoundaryName>("secondary_boundary") = _secondary;
+        params.set<SubdomainName>("primary_subdomain") = primary_subdomain_name;
+        params.set<SubdomainName>("secondary_subdomain") = secondary_subdomain_name;
+        params.set<bool>("use_displaced_mesh") = true;
+        params.set<Real>("c_t") = getParam<Real>("c_tangential");
+        params.set<Real>("c") = getParam<Real>("c_normal");
+        params.set<bool>("compute_primal_residuals") = false;
+
+        params.set<std::vector<VariableName>>("disp_x") = {displacements[0]};
+        if (ndisp > 1)
+          params.set<std::vector<VariableName>>("disp_y") = {displacements[1]};
+
+        params.set<NonlinearVariableName>("variable") = normal_lagrange_multiplier_name;
+        params.set<std::vector<VariableName>>("friction_lm") = {
+            tangential_lagrange_multiplier_name};
+        params.set<Real>("mu") = getParam<Real>("friction_coefficient");
+        // secondary_disp_z is not implemented for tangential (yet).
+
+        _problem->addConstraint(
+            "ComputeFrictionalForceLMMechanicalContact", action_name + "_tangential_lm", params);
```

Comment:
This could probably be simplified as there is quite a bit of shared code between the coulomb/!coulomb branches (i.e. swap the nesting order of the conditionals)

Reply:
never mind, not worth the hassle